src/get-configvals.c

```diff
@@ -0,0 +1,44 @@
+/* 
+* Copyright (c) 2017, SingularityWare, LLC., Inc. All rights reserved.
+* 
+* This software is licensed under a 3-clause BSD license.  Please
+* consult LICENSE file distributed with the sources of this project regarding
+* your rights to use or distribute this software.
+* 
+*/
+
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <string.h>
+
+#include "util/util.h"
+#include "util/config_parser.h"
+
+#ifndef SYSCONFDIR
+#error SYSCONFDIR not defined
+#endif
+
+int main(int argc, char **argv) {
+
+    if ( argc < 2 ) {
+        printf("USAGE: %s [key]\n", argv[0]);
+        exit(0);
+    }
+
+    char *key = strdup(argv[1]);
```

Comment:
Declare your vars at the top of the scope. In this case : char *key; at line 23.
Then here (29) just : key = strdup(argv[1]);

Now if _singularity_config_get_value_impl() doesn't change the content of the string at "key", you could just pass argv[1] to _singularity_config_get_value_impl() and avoid the declaration of key, the strdup() and the free() of this intermediate var.